server/src/main/java/org/apache/druid/segment/realtime/plumber/Committers.java

```diff
@@ -40,54 +42,15 @@ public void run()
       // Do nothing
     }
   };
+  private static final Supplier<Committer> NIL_SUPPLIER = Suppliers.ofInstance(NIL);
```

Comment:
It seems like a smell that this class still exists and only has this nil supplier, and maybe can be removed entirely. But I'm ok if we continue clean-up in the future since this looks a bit more invasive to dump than just removing `commit` from the `Firehose`.

Reply:
Agree that we should remove the nil supplier. A follow up PR is fine as long as we remember to do it...